services/user.py

```diff
@@ -0,0 +1,59 @@
+from django.db import transaction
+
+from db.models import User
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None,
+) -> User:
+    with transaction.atomic():
```

Comment:
You could apply a `transaction.atomic` decorator to this function so that all function body would be treated as atomic transaction.